lib/CL/devices/basic/basic.c

```diff
@@ -960,3 +956,102 @@ pocl_basic_set_kernel_exec_info_ext (cl_device_id dev,
       return CL_INVALID_VALUE;
     }
 }
+
+static int
+get_dbk_index (cl_program p, cl_kernel k)
+{
+
+  int dbk_index = -1;
+  for (int i = 0; i < p->num_builtin_kernels; ++i)
+    {
+      if (strcmp (p->builtin_kernel_names[i], k->name) == 0)
+        {
+          dbk_index = i;
+          return dbk_index;
+        }
+    }
+  return dbk_index;
+}
+
+/**
+ * CPU_DBK_TAG
+ */
+int
+pocl_basic_create_kernel (cl_device_id device,
+                          cl_program p,
+                          cl_kernel k,
+                          unsigned device_i)
+{
+
+  /* no dbks, nothing to do */
```

Comment:
Extract this functionality to a separate helper method so we can call it from other device drivers which want to add JPEG DBK support.

Reply:
For now it's specific to cpu dbks. I'd imagine that other devices could have a different method of creating their kernels.